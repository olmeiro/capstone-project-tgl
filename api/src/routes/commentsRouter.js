const express = require("express");
const router = express.Router();

const {
  getCommentsByPost,
  uploadComment,
  putComment,
  deleteComment,
} = require("../controllers/commentsController");

router.get("/bypost/:postid", getCommentsByPost);
router.post("/", uploadComment);
router.put("/", putComment);
router.delete("/:commentId", deleteComment);

module.exports = router;
