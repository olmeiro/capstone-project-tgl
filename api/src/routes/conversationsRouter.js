const express = require("express");
const router = express.Router();

const {
    getConversationsByUserId,
    postConversation
} = require("../controllers/conversationsController");


router.get("/:userId", getConversationsByUserId);
router.post("/", postConversation)

module.exports = router