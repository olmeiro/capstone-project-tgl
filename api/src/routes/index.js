const express = require("express");
const router = express.Router();


const userRouter = require("./userRouter");
const postsRouter = require("./postsRouter")
const commentsRouter = require("./commentsRouter");
const friendsRouter = require("./friendsRouter");
const favoritesRouter = require("./favoritesRouter");
const conversationsRouter = require("./conversationsRouter")
const messagesRouter = require("./messagesRouter")

router.use("/user", userRouter)
router.use("/posts", postsRouter)
router.use("/comments", commentsRouter)
router.use("/friends", friendsRouter)
router.use("/favorites", favoritesRouter)
router.use("/conversations", conversationsRouter)
router.use("/messages", messagesRouter)

module.exports = router