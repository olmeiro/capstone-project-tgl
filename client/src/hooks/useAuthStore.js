import React, { useDispatch, useSelector } from 'react-redux'
import { socialApi } from '../api'
import { clearErrorMessage, onChecking, onLogin, onLogout, onLoadDataProfile } from '../store'

export const useAuthStore = () => {
  const { status, user, errorMessage } = useSelector(state => state.auth)

  const dispatch = useDispatch()

  const startLogin = async ({ alias, password }) => {
    dispatch(onChecking())
    try {
      const { data } = await socialApi.post('/user/login', { alias, password })
      localStorage.setItem('token', data.body.token)

      const id = data.body.user.id
      const aliasUser = data.body.user.alias
      const name = data.body.user.name
      const bio = data.body.user.bio
      const email = data.body.user.email
      const phone = data.body.user.phone
      const photoProfile = data.body.user.photoProfile
      const favorites = data.body.user.favorites
      const friends = data.body.user.friends

      dispatch(onLoadDataProfile({ id, aliasUser, name, bio, email, phone, photoProfile, favorites, friends }))
      dispatch(onLogin({ id: data.body.user.id, alias: data.body.user.alias, name: data.body.user.name, photoProfile: data.body.user.photoProfile, friends: data.body.user.friends }))
    } catch (error) {
      dispatch(onLogout(error.response.data?.message || ''))
      setTimeout(() => {
        dispatch(clearErrorMessage())
      }, 10)
    }
  }

  const startRegister = async ({ alias, name, email, phone, password }) => {
    dispatch(onChecking())
    try {
      const { data } = await socialApi.post('/user', { alias, name, email, phone, password })
      localStorage.setItem('token', data.body.token)
      dispatch(onLogin({ id: data.body.user.id, alias: data.body.user.alias, name: data.body.user.name }))
    } catch (error) {
      dispatch(onLogout(error.response.data?.message || ''))
      setTimeout(() => {
        dispatch(clearErrorMessage())
      }, 10)
    }
  }

  const checkToken = async () => {
    const token = localStorage.getItem('token')
    if (!token) {
      return dispatch(onLogout())
    }

    try {
      const { data } = await socialApi('/user/renew')
      localStorage.setItem('token', data.body.token)

      const dataUser = await socialApi.get(`/user/byid/${data.body.id}`)

      const id = dataUser.data.body.id
      const aliasUser = dataUser.data.body.alias
      const name = dataUser.data.body.name
      const bio = dataUser.data.body.bio
      const email = dataUser.data.body.email
      const phone = dataUser.data.body.phone
      const photoProfile = dataUser.data.body.photoProfile
      const favorites = dataUser.data.body.favorites
      const friends = dataUser.data.body.friends

      dispatch(onLoadDataProfile({ id, aliasUser, name, bio, email, phone, photoProfile, favorites, friends }))
      dispatch(onLogin({ id: data.body.id, alias: data.body.alias, name: data.body.name }))
    } catch (error) {
      localStorage.clear()
      dispatch(onLogout())
    }
  }

  const startLogout = () => {
    localStorage.clear()
    dispatch(onLogout())
  }

  return {
    // properties
    errorMessage,
    status,
    user,
    // methods
    startLogin,
    startRegister,
    checkToken,
    startLogout
  }
}
