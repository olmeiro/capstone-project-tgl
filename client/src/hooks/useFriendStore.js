import { useDispatch, useSelector } from 'react-redux'
import { socialApi } from '../api'
import { getFriendsFromUser, deleteFriend } from '../store'

export const useFriendStore = () => {
  const dispatch = useDispatch()
  const { user } = useSelector(state => state.auth)
  const userId = user.id

  const getFriendsFromUserHook = async () => {
    const response = await socialApi.get(`/friends/${userId}`)
    const friends = response.data.body
    dispatch(getFriendsFromUser(friends))
  }
  const deleteFriendHook = async friendId => {
    await socialApi.delete(`/friends?userid=${userId}&friendid=${friendId}`)
    dispatch(deleteFriend(friendId))
  }
  return {
    getFriendsFromUserHook,
    deleteFriendHook
  }
}
