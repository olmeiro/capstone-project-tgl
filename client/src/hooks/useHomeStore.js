import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { socialApi } from '../api'
import { checkFavorites, getFavorites, actionKeepLastUserVisitesOnRedux, getPostsToHome, getFriendsFromFriends, checkComments, getUserSearched, checkEmptySearchBar, setPathReference, setLastUserVisited, setInfoFromUserLoggedIn, setcheckInfoUser, setPostsFromUserLoggedIn } from '../store'

export const useHomeStore = () => {
  const dispatch = useDispatch()
  const { user } = useSelector(state => state.auth)
  const userId = user.id

  const entire = socialApi.get(`/user/byid/${userId}`).then(response => response.data.body)
  let friends = entire.then(user => user.friends)

  const getPostsToHomeHook = async () => {
    friends = await friends
    let postsTotalFromFriends = friends.map(async friendId => await socialApi.get(`/posts/byuser/${friendId}`))
    postsTotalFromFriends = Promise.all(postsTotalFromFriends)
    postsTotalFromFriends = await postsTotalFromFriends
    postsTotalFromFriends = postsTotalFromFriends.map(posts => posts.data.body).flat()
    dispatch(getPostsToHome(postsTotalFromFriends))
  }
  const getFriendsFromFriendsHook = async () => {
    const wholeUser = await socialApi.get(`/user/byid/${userId}`)
    const friendsId = wholeUser.data.body.friends
    let suggestion = friendsId.map(async id => {
      const response = await socialApi.get(`/user/byid/${id}`)
      const data = response.data
      const suggestion = data.body.friends
      return suggestion
    })
    suggestion = Promise.all(suggestion)
    suggestion = await suggestion
    suggestion = suggestion.flat()
    suggestion = suggestion.filter((friendid, index) => suggestion.indexOf(friendid) == index)
    suggestion = suggestion.map(async id => {
      const response = await socialApi.get(`/user/byid/${id}`)
      const data = response.data
      const suggestion = data.body
      return suggestion
    })
    suggestion = Promise.all(suggestion)
    suggestion = await suggestion
    dispatch(getFriendsFromFriends(suggestion))
  }
  const addFriendshipHook = async (friendId) => {
    await socialApi.post('/friends/', { userId, friendId })
  }
  const likeAPost = async (id) => {
    await socialApi.put('/posts/', { id, likes: 3 })
  }
  const makeAComment = async (postId, comment) => {
    await socialApi.post('/comments/', { comment, postId, userId })
  }
  const getCommentsByPost = async (postid) => {
    let comments = await socialApi.get(`/comments/bypost/${postid}`)
    comments = comments.data.body
    comments = comments.map(async comment => {
      const userId = comment.UserId
      let user = await socialApi.get(`/user/byid/${userId}`)
      user = user.data.body
      comment.user = user
      return comment
    })
    comments = await Promise.all(comments)
    return comments
  }
  const deleteComment = async (commentId) => {
    await socialApi.delete(`/comments/${commentId}`)
  }
  const checkCommentsHook = () => {
    dispatch(checkComments())
  }
  const searchUserByAlias = async (aliasFromBody) => {
    let foundUsers = null
    if (aliasFromBody.trim() != '') {
      const response = await socialApi.get('/user/all')
      const users = response.data.body
      foundUsers = users.filter(user => user.alias.toLowerCase().includes(aliasFromBody && aliasFromBody.toLowerCase()))
    }
    dispatch(getUserSearched(foundUsers))
  }
  const checkEmptySearchBarHook = () => {
    dispatch(checkEmptySearchBar())
  }
  const sendPathHook = (pathReference) => {
    dispatch(setPathReference(pathReference))
  }
  const setLastUserVisitedHook = async (userAlias, userId) => {
    const responseUser = await socialApi.get(`user/byalias/${userAlias}`)
    const user = responseUser.data.body
    const responsePosts = await socialApi.get(`/posts/byuser/${userId}`)
    const posts = responsePosts.data.body
    user.posts = posts
    localStorage.setItem('lastUserVisited', JSON.stringify(user))
    const lastUserVisited = JSON.parse(localStorage.getItem('lastUserVisited'))
    dispatch(setLastUserVisited(lastUserVisited))
  }
  const keepLastUserVisitesOnRedux = async () => {
    const lastUserVisited = JSON.parse(localStorage.getItem('lastUserVisited'))
    dispatch(actionKeepLastUserVisitesOnRedux(lastUserVisited))
  }
  const getInfoFromTheUserLoggedIn = async () => {
    const user = await entire
    const responsePosts = await socialApi.get(`/posts/byuser/${userId}`)
    const posts = responsePosts.data.body
    const newUser = {
      ...user,
      posts
    }
    dispatch(setInfoFromUserLoggedIn(newUser))
  }
  const checkInfoUserHook = async () => {
    dispatch(setcheckInfoUser())
  }
  const getPostsFromUserLoggedIn = async () => {
    const responsePosts = await socialApi.get(`/posts/byuser/${userId}`)
    const posts = responsePosts.data.body
    dispatch(setPostsFromUserLoggedIn(posts))
  }
  const addToFavorites = async (postFavoriteId, userLogeadoId) => {
    await socialApi.post(`/favorites?postId=${postFavoriteId}&userId=${userLogeadoId}`)
  }
  const getFavoritesHook = async () => {
    const response = await socialApi.get(`/favorites/${userId}`)
    const favorites = response.data.body
    dispatch(getFavorites(favorites))
  }
  const deleteFavoriteHook = async (postId) => {
    await socialApi.delete(`/favorites?postId=${postId}&userId=${userId}`)
  }
  const checkFavoritesHook = async () => {
    dispatch(checkFavorites())
  }

  return {
    getPostsToHomeHook,
    getFriendsFromFriendsHook,
    addFriendshipHook,
    likeAPost,
    makeAComment,
    getCommentsByPost,
    deleteComment,
    checkCommentsHook,
    searchUserByAlias,
    checkEmptySearchBarHook,
    sendPathHook,
    setLastUserVisitedHook,
    getInfoFromTheUserLoggedIn,
    checkInfoUserHook,
    getPostsFromUserLoggedIn,
    keepLastUserVisitesOnRedux,
    addToFavorites,
    getFavoritesHook,
    deleteFavoriteHook,
    checkFavoritesHook
  }
}
