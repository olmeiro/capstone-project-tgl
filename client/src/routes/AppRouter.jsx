import React, { useEffect } from 'react'
import { Navigate, Route, Routes } from 'react-router-dom'

import { useAuthStore } from '../hooks'
import LoadingSpinner from '../components/loading/LoadingSpinner'
import { HomePage } from '../components/home/pages/HomePage'
import { ProfilePage } from '../components/home/pages/ProfilePage'
import { Friends } from '../components/home/pages/Friends'
import { LoginPage } from '../components/auth/pages/LoginPage'
import { RegisterPage } from '../components/auth/pages/RegisterPage'

import { ProfilePageOfOneUser } from '../components/home/pages/ProfilePageOfOneUser'
import { Favorites } from '../components/home/pages/Favorites'
import { Messenger } from '../components/home/pages/Messenger'

export const AppRouter = () => {
  const { status, checkToken } = useAuthStore()

  useEffect(() => {
    checkToken()
  }, [])

  if (status === 'checking') {
    return (
      <LoadingSpinner />
    )
  }

  return (
    <div className='bg-purple-100'>
      <Routes>
        {
          (status === 'not-authenticated')
            ? (
              <>
                <Route path="/auth/login" element={<LoginPage />} />
                <Route path="/auth/register" element={<RegisterPage />} />

              <Route path='/*' element={<Navigate to="/auth/login" />} />
            </>
              )
            : (
            <>
              <Route path='/' element={<HomePage />} />
              <Route path='/profile' element={<ProfilePage />} />
              <Route path='/friends' element={<Friends />} />
              <Route path='/profile/:userAlias' element={<ProfilePageOfOneUser />} />
              <Route path='/favorites' element={<Favorites />} />
              <Route path='/messenger' element={<Messenger />} />

                <Route path='/*' element={<Navigate to="/" />} />
              </>
              )
        }
      </Routes>
    </div>
  )
}
