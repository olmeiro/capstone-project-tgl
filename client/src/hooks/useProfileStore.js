import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import Swal from 'sweetalert2'

import { socialApi } from '../api'
import {
  onLoadDataProfile,
  onChangeDataProfile,
  onLoadCommentPhoto,
  inactivatingCount,
  deletingCount,
  onLoadFriendsUser,
  onLoadPublication,
  onSendPublication,
  onLoadPhotoProfile,
  onChanging
} from '../store'
import { useHomeStore } from './useHomeStore'

export const useProfileStore = () => {
  const profile = useSelector((state) => state.profile)
  const dispatch = useDispatch()
  const { getPostsFromUserLoggedIn } = useHomeStore()
  const loadingDataProfile = async (id) => {
    try {
      const { data } = await socialApi.get(`/user/byid/${id}`)
      dispatch(onLoadDataProfile(data.body))
      Swal.fire({
        icon: 'success',
        title: 'Imagen cargada correctamente.'
      })
    } catch (error) {
      Swal.fire({
        icon: 'error',
        title: 'Oops...',
        text: 'Algo ha ido mal!'
      })
    }
  }

  const loadingFriendsUser = () => {
    // get info api
    dispatch(onLoadFriendsUser()) // pasar info api
  }

  const loadingPhotoProfile = async (image, idUser) => {
    const formData = new FormData()
    formData.append('file', image)
    try {
      await socialApi.put('/user/profilephoto', formData)
      let user = await socialApi.get(`/user/byid/${idUser}`)
      user = user.data.body
      dispatch(onLoadPhotoProfile(user.photoProfile))
      Swal.fire({
        icon: 'success',
        title: 'Fotos de perfil cargada correctamente.'
      })
    } catch (error) {
      Swal.fire({
        icon: 'error',
        title: 'Oops...',
        text: 'Algo ha ido mal!'
      })
    }
  }

  const sendPublicationUser = async (loginUserId, comment, image) => {
    const formData = new FormData()
    formData.append('loginUserId', loginUserId)
    formData.append('description', comment)
    formData.append('file', image)

    try {
      onLoadChanging()
      const { data } = await socialApi.post('/posts', formData, {
        headers: {
          'Content-Type': 'multipart/form-data'
        }
      })
      getPostsFromUserLoggedIn()
      dispatch(onSendPublication(data.body))
      onLoadChanging()
      Swal.fire({
        icon: 'success',
        title: 'Fotos cargadas correctamente.'
      })
    } catch (error) {
      Swal.fire({
        icon: 'error',
        title: 'Oops...',
        text: 'Algo a ido mal cargando las imagenes!'
      })
    }
  }

  const loadingPublicationUser = async (id) => {
    try {
      const { data } = await socialApi.get('/posts/all', id)
      dispatch(onLoadPublication(data.body))
    } catch (error) {
      Swal.fire({
        icon: 'error',
        title: 'Oops...',
        text: 'Algo a ido mal cargando las publicaciones!'
      })
    }
  }

  const changeDataProfile = async ({
    idUser,
    alias,
    name,
    bio,
    email,
    phone
  }) => {
    try {
      await socialApi.put('/user/', { idUser, alias, name, bio, email, phone })
      dispatch(onChangeDataProfile({ alias, name, bio, email, phone }))
      Swal.fire({
        icon: 'success',
        title: 'Información actualizada correctamente.'
      })
    } catch (error) {
      Swal.fire({
        icon: 'error',
        title: 'Información no actualizada correctamente.'
      })
    }
  }

  const uploadCommentPhoto = (comment, id) => {
    // TODO: hacer put comment en photo cover de user id
    dispatch(onLoadCommentPhoto(comment))
  }

  const deletePostUser = async (postId, userId) => {
    try {
      Swal.fire({
        title: 'Esta seguro de eliminar publicación?',
        text: 'No puedes revertir esta acción!',
        icon: 'warning',
        showCancelButton: true,
        confirmButtonColor: '#3085d6',
        cancelButtonColor: '#d33',
        confirmButtonText: 'Sí, borrar!'
      }).then(async (result) => {
        if (result.isConfirmed) {
          try {
            await socialApi.delete(`/posts/${postId}`)
            getPostsFromUserLoggedIn()
            loadingPublicationUser(userId)
            Swal.fire(
              'Borrado!',
              'La publicación ha sido eliminada.',
              'success'
            )
          } catch (error) {
            Swal.fire('No se pudo eliminar la publicación.')
          }
        }
      })
    } catch (error) {
      Swal.fire({
        icon: 'error',
        title: 'Información no actualizada correctamente.'
      })
    }
  }

  const inactiveCount = (id) => {
    // TODO: change data api user id borrado logico
    dispatch(inactivatingCount(id))
    // onLogout auth
  }

  const deleteCount = (id) => {
    // TODO: change data api user id
    dispatch(deletingCount(id))
  }

  const onLoadChanging = () => {
    dispatch(onChanging())
  }

  return {
    // properties
    profile,
    loadingFriendsUser,
    loadingPhotoProfile,
    sendPublicationUser,
    loadingPublicationUser,
    // methods
    onLoadChanging,
    loadingDataProfile,
    changeDataProfile,
    uploadCommentPhoto,
    deletePostUser,
    inactiveCount,
    deleteCount
  }
}
