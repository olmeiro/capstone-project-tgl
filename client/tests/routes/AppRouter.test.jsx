import { render, screen } from '@testing-library/react'
import * as reactRedux from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import { useAuthStore } from '../../src/hooks/useAuthStore'
import { AppRouter } from '../../src/routes/AppRouter'

jest.mock('../../src/hooks/useAuthStore')
jest.mock('../../src/hooks/useHomeStore')
jest.mock('../../src/components/home/pages/HomePage', () => ({
  HomePage: () => <h1>Home Page</h1>
}))

describe('Test <AppRouter/>', () => {
  const mockCheckAuthToken = jest.fn()
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('should render Loading component and call checkToken', () => {
    useAuthStore.mockReturnValue({

      status: 'checking',
      checkToken: mockCheckAuthToken
    })

    render(
        <AppRouter />
    )
    // screen.debug()
    expect(screen.findAllByText('spinner-container')).toBeTruthy()
    expect(mockCheckAuthToken).toHaveBeenCalled()
  })

  test('should render component LoginPage when not-authenticated', () => {
    useAuthStore.mockReturnValue({
      errorMessage: undefined,
      status: 'not-authenticated',
      user: {},
      checkToken: jest.fn()
    })

    const { container } = render(
      <MemoryRouter>
        <AppRouter />
      </MemoryRouter>
    )
    const loginBtn = screen.getByLabelText('btn-login')

    expect(loginBtn).toBeTruthy()
    expect(container).toMatchSnapshot()
  })

  test('should render component HomePage when authenticated', () => {
    useAuthStore.mockReturnValue({
      errorMessage: undefined,
      status: 'authenticated',
      checkToken: mockCheckAuthToken
    })

    render(
        <MemoryRouter >
          <AppRouter />
        </MemoryRouter>
    )
    // screen.debug()
    expect(screen.getByText('Home Page')).toBeTruthy()
  })
})
