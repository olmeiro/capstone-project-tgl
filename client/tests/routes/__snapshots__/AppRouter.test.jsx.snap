// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AppRouter/> should render component LoginPage when not-authenticated 1`] = `
<div>
  <div
    class="bg-purple-100"
  >
    <section
      class="w-full  flex flex-row sm:flex-col min-h-screen"
    >
      <header
        class="w-full min-h-full flex flex-row sm:flex-col md:flex-row justify-between p-6 items-center h-28 bg text-team-green"
      >
        <div
          class="mt-2"
        >
          <img
            alt="logo team"
            class="w-24"
            src=""
          />
        </div>
      </header>
      <main
        class="flex flex-row min-h-[74.2vh] sm:flex-col md:flex-row"
      >
        <section
          class="w-1/2 sm:w-full flex flex-col justify-center p-10 place-items-center"
        >
          <h1
            class="text-7xl sm:text-xl p-6"
          >
            Red Social TGL
          </h1>
          <img
            alt="logo team"
            class="bg w-56 rounded-lg p-9 mt-9 mb-9"
            src=""
          />
          <footer>
            <p
              class="text-sm text-clip whitespace-normal"
            >
              Red social Top Gun Lab. Realiza tus proyectos en compañia de los mejores.
            </p>
          </footer>
        </section>
        <section
          class="w-1/2 sm:w-full flex flex-col justify-center border-b-team-green"
        >
          <h4
            class="text-center text-5xl p-3 text-team-blue  "
          >
            Login
          </h4>
          <form
            aria-label="form"
            class="flex flex-col p-8"
          >
            <label
              for=""
            >
              Correo electrónico
            </label>
            <input
              class="form-input px-4 py-3 rounded-full"
              data-testid="data-alias"
              id="alias"
              name="alias"
              placeholder="alias"
              type="text"
              value=""
            />
            <span
              class="text-[10px] text-end text-team-blue"
            />
            <label
              for=""
            >
              Contraseña
            </label>
            <input
              class="form-input px-4 py-3 rounded-full"
              data-testid="data-pw"
              id="password"
              name="password"
              placeholder="password"
              type="password"
              value=""
            />
            <span
              class="text-[10px] text-end text-team-blue"
            />
            <button
              aria-label="btn-login"
              class="bg-team-brown h-10 w-full  ml-3 mt-2 rounded-lg text-team-dark font-medium hover:bg-team-green"
              data-testid="btn-login"
              type="submit"
            >
              Login
            </button>
            <div
              class="flex flex-row-reverse justify-between mt-4"
            >
              <a
                class="flex justify-center place-items-center underline text-team-blue  hover:bg-gray h-10 w-1/2 rounded-lg font-medium "
                color="inherit"
                data-testid="create-count"
                href="/auth/register"
              >
                Crear cuenta
              </a>
            </div>
          </form>
        </section>
      </main>
      <footer
        class="max-w-full mt-[10] bg-team-blue h-20 text-lg font-medium text-slate-200 flex text-center justify-center items-center"
      >
        <p>
          ©2022 tgl_group. All Rights Reserved.
        </p>
      </footer>
    </section>
  </div>
</div>
`;
